typescript/nomad-deploy/src/core/index.ts

```diff
@@ -676,6 +676,132 @@ export async function deployNChains(deploys: CoreDeploy[]) {
   }
 }
 
+
+/**
+ * Deploy the entire suite of Nomad contracts
+ * on each chain within the chainConfigs array
+ * including the upgradable Home, Replicas, and GovernanceRouter
+ * that have been deployed, initialized, and configured
+ * according to the deployNomad script
+ *
+ * @dev The first chain in the array will be the governing chain
+ *
+ * @param deploys - An array of chain deploys
+ */
+export async function deployHubAndSpoke(hub: CoreDeploy, spokes: CoreDeploy[]) {
+  if (!hub) {
+    throw new Error('Must pass hub config');
+  } else if (spokes.length == 0) {
+    throw new Error('Must pass at least one spoke config');
+  }
+
+  const deploys = [hub, ...spokes];
+  const valueArr = deploys.map(function (item) {
+    return item.chain.domain;
+  });
+  const duplicateDomains = valueArr.some(function (item, idx) {
+    return valueArr.indexOf(item) != idx;
+  });
+  if (duplicateDomains) {
+    throw new Error(
+        'You have specified multiple deploys with the same domain. Check your config.',
+    );
+  }
+
+  // there exists any chain marked test
+  const isTestDeploy: boolean = deploys.filter((c) => c.test).length > 0;
+
+  log(isTestDeploy, `Beginning 1 Hub, ${spokes.length} Spoke${spokes.length > 1 ? "s" : ""} deploy process`);
+  log(isTestDeploy, `Deploy env is ${hub.config.environment}`);
+  log(isTestDeploy, `${hub.chain.name} is governing hub`);
+  log(isTestDeploy, `${JSON.stringify(spokes.map(deploy => deploy.chain.name))} ${spokes.length > 1 ? "are spokes" : "is spoke"}`);
+  deploys.forEach((deploy) => {
+    log(
+        isTestDeploy,
+        `Updater for ${deploy.chain.name} Home is ${deploy.config.updater}`,
+    );
+  });
+
+  if(!isTestDeploy && !hub.config.governor) {
+    throw new Error(
+        `Hub has no governor config`,
+    );
+  }
+
+  log(isTestDeploy, 'awaiting provider ready');
+  await Promise.all([
+    deploys.map(async (deploy) => {
+      await deploy.ready();
+    }),
+  ]);
+  log(isTestDeploy, 'done readying');
+
+  // store block numbers for each chain, so that agents know where to start
+  await Promise.all(deploys.map((d) => d.recordFromBlock()));
+
+  // deploy nomad on each chain
+  await Promise.all(
+      deploys.map(async (deploy) => {
+        await deployNomad(deploy);
+      }),
+  );
+
+  // enroll remotes on every chain
+  //
+  //    NB: do not use Promise.all for this block. It introduces a race condition
+  //    which results in multiple replica implementations on the home chain.
+  //
```

Comment:
You should use `Promise.all` for enrolling the Spokes' Replicas of the Hub's Home

Reply:
https://github.com/nomad-xyz/nomad-monorepo/pull/3/commits/30a0b0d23e7171f6c01fc89835659545bca08415